content/blog/package-react-components/index.md

```diff
@@ -0,0 +1,776 @@
+but all the Assets that might be related to that component such as CSS, Images, SVGs and fonts.
+These assets are directly related to our Javascript code via some of the following statements:
+
+- `imports image from './image.{png,jpg,svg}'` in Javascript
+- `import classNames from './styles.{css,scss,less)'` in Javascript
+- `url(image.{png,jpg,svg})` in CSS
+- `font-family and @import` in CSS
+
+[Webpack][2] can understand all these dependency statements via _loaders_ and completely
+analyse the cross-language and cross-file-type dependency graph, which for a very simple
+component might look something like this.
+
+```
+            --> styles.css --> otherImage.svg
+MyComponent --> image.svg
+
+```
+
+It turns out that there are a lot of schemes for representing this graph in a Packaged NPM module,
+each with it's own trade-offs. Analysing each alternative is the main purpose of this post.
+
+Unfortunately in early 2020 there still no universal standard way of Packaging a Front End module with all
+its assets, but a lot of progress has been made and there are actually lots of promising solutions.
+
+We are going to explore how this problem is tackled in the wild by studying
+each technique and providing example real world production ready libraries that
+use them and in the end of this post we are going
+to recommend a particular solution to this problem based of my experience.
+
+Additionally we are going to explore how each of these packaging schemes enable
+different **Theming** schemes and, at the end, recommend one.
+
+Most of this post is also applicable to any piece of Front End Javascript code that is related
+to CSS, images, etc, such as [Angular][3] Components / Directives, [Vue][15] Components, etc.
+
+## To Inline or not to Inline
+
+In the context of Front End assets, inlining means to bake those assets directly
+into the Javascript code (bundle) generated by our bundlers or compilers such as Webpack or [Babel][4].
+
+CSS can be inlined without any third party library but to enable all regular CSS features
+a third party **CSS-in-JS** library is recommended such as [Emotion][5] or [Styled-components][6], more on this later.
+Images can be inlined by Webpack's [url-loader][11] in the form of [Data URL][100] with base 64 encoding.
+SVG can be inlined by turning an `.svg` file into a [React][1] Component (An SVG file is just a regular HTML tree)
+or by using Webpack's [inline-svg-loader][101]
+
+The main advantage of inlining is that after you bundle all your code base you end up with
+a single Javascript file that contains all the dependencies it needs to run, even CSS and Images.
+
+This makes _very_ easy to package and publish to [NPM][102] and it makes very easy
+for **Component Consumers** to _use_ your module.
+
+Using **CSS-in-JS** libraries like [Emotion][5] or [Styled-components][6] makes up for a very
+nice **Run-Time** **Theming** API. Since everything is Javascript your Themes are regular Javascript
```

Comment:
This clears up some confusion I had with what theming actually meant and that I expressed above.

Reply:
nit for reading clarity: I think I would put a comma so that it says "Since everything is Javascript**,** your themes are regular Javascript"